notus/scanner/models/os/euleros.py

```diff
@@ -0,0 +1,68 @@
+# Copyright (C) 2021-2022 Greenbone Networks GmbH
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import re
+from typing import List, Optional
+from paramiko import SSHClient
+from .distribution import Distribution
+
+# EulerOS 2.0
+_euler_re = re.compile(
+    r"EulerOS release ([0-9]+\.[0-9]+)\s?(\((SP[0-9]+)(x86_64)?\))?"
+)
+# EulerOS Virtualzations
+_euler_v_re = re.compile(r"EulerOS Virtualization ([a-zA-Z0-9 ]+) ([0-9.]+)")
+
+
+class EulerOS(Distribution):
+    @staticmethod
+    def get_os(ssh: SSHClient) -> Optional[str]:
+        # Try EulerOS
+        os = None
+        rls = EulerOS.command(ssh, "cat /etc/euleros-release")
+        if rls.find("EulerOS release") >= 0:
+            # Euler OS release 2.0 ...
+            rls_match = _euler_re.match(rls).groups()
+            if rls_match[0]:
+                os = "euleros_v" + rls_match[0]
+            if rls_match[2]:
```

Comment:
That could potentially result in an `tuple index out of range` error, please verify length of groups before accessing it. Otherwise one could run into issues when the regex just matches partly or it got changed.